extensions/data-transfer/portability-data-transfer-apple/src/main/java/org/datatransferproject/datatransfer/apple/photos/ApplePhotosImporter.java

```diff
@@ -0,0 +1,112 @@
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.datatransferproject.datatransfer.apple.photos;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.datatransferproject.api.launcher.Monitor;
+import org.datatransferproject.datatransfer.apple.AppleInterfaceFactory;
+import org.datatransferproject.datatransfer.apple.constants.ApplePhotosConstants;
+import org.datatransferproject.datatransfer.apple.constants.ApplePhotosConstants.AppleMediaType;
+import org.datatransferproject.spi.transfer.idempotentexecutor.IdempotentImportExecutor;
+import org.datatransferproject.spi.transfer.provider.ImportResult;
+import org.datatransferproject.spi.transfer.provider.Importer;
+import org.datatransferproject.transfer.JobMetadata;
+import org.datatransferproject.types.common.models.media.MediaAlbum;
+import org.datatransferproject.types.common.models.photos.PhotosContainerResource;
+import org.datatransferproject.types.transfer.auth.AppCredentials;
+import org.datatransferproject.types.transfer.auth.TokensAndUrlAuthData;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * An Apple importer to import the Photos into Apple iCloud-photos.
+ */
+public class ApplePhotosImporter
+    implements Importer<TokensAndUrlAuthData, PhotosContainerResource> {
+  private final AppCredentials appCredentials;
+  private final String exportingService;
+  private final Monitor monitor;
+  private final AppleInterfaceFactory factory;
+
+  public ApplePhotosImporter(
+      @NotNull final AppCredentials appCredentials, @NotNull final Monitor monitor) {
+    this(appCredentials, JobMetadata.getExportService(), monitor, new AppleInterfaceFactory());
+  }
+
+  @VisibleForTesting
+  ApplePhotosImporter(
+      @NotNull final AppCredentials appCredentials,
+      @NotNull final String exportingService,
+      @NotNull final Monitor monitor,
+      @NotNull  AppleInterfaceFactory factory) {
+    this.appCredentials = appCredentials;
+    this.exportingService = exportingService;
+    this.monitor = monitor;
+    this.factory = factory;
+  }
+
+  @Override
+  public ImportResult importItem(
+      UUID jobId,
+      IdempotentImportExecutor idempotentImportExecutor,
+      TokensAndUrlAuthData authData,
+      PhotosContainerResource data)
+      throws Exception {
+    if (data == null) {
+      return ImportResult.OK;
+    }
+
+    AppleMediaInterface mediaInterface = getMediaInterface(authData);
```

Comment:
[for a future PR] optional, but something I recently added when it comes to passing this auth data around:

```suggestion
    // WARNING: leave this here inside importItem context, and do _not_ store this as a member on this importer's class, as it contains auth data that's dependent on this particular request for one user.
    AppleMediaInterface mediaInterface = getMediaInterface(authData);
```

Reply:
related: another defensive move would be to have your factory[1] take in the jobId so that no to requests to the same importer instance can accidentally work with the same interface

[1]: (and update your factory to store the interface on a map of ids)